src/OpenTelemetry.Collector.Dependencies/DependenciesCollector.cs

```diff
@@ -40,7 +40,11 @@ public DependenciesCollector(DependenciesCollectorOptions options, ITracer trace
         {
             this.diagnosticSourceSubscriber = new DiagnosticSourceSubscriber(
                 new Dictionary<string, Func<ITracer, Func<HttpRequestMessage, ISampler>, ListenerHandler>>()
-                { { "HttpHandlerDiagnosticListener", (t, s) => new HttpHandlerDiagnosticListener(t, s) } },
+                {
+                    { "HttpHandlerDiagnosticListener", (t, s) => new HttpHandlerDiagnosticListener(t, s) },
+                    { "Azure.Clients", (t, s) => new AzureSdkDiagnosticListener("Azure.Clients", t, sampler) },
```

Comment:
`s` is not necessarily the same `sampler` - sampler could be null and this crazy lambda falls back to something. So please use `s`

Reply:
s is lambda that takes HttpRequestMessage, what do I pass into it?

Reply:
no I mean this lambda:

```csharp
x =>
                {
                    ISampler s = null;
                    try
                    {
                        s = options.CustomSampler(x);
                    }
                    catch (Exception e)
                    {
                        s = null;
                        DependenciesCollectorEventSource.Log.ExceptionInCustomSampler(e);
                    }
                    return s ?? sampler;
                    });
```

I.e. `"Azure.Clients", (t, s) => new AzureSdkDiagnosticListener("Azure.Clients", t, sampler)` -> `"Azure.Clients", (t, s) => new AzureSdkDiagnosticListener("Azure.Clients", t, s)`

Reply:
Right, type of s is `Func<HttpRequestMessage, ISampler>` how do I use it in my listener to get a sampler? Always pass null in?

Reply:
oh, I see. sorry.  let me think for a moment

Reply:
how about this? https://github.com/open-telemetry/opentelemetry-dotnet/pull/170